File: archive/string-matching-in-an-array.js
const stringMatching = function(words) {
  let ans = []
  for (let i = 0; i < words.length; i++) {
    for (let j = 0; j < words.length; j++) {
      if (i === j) continue
      if (words[j].split(words[i]).join('').length !== words[j].length) {
        ans.push(words[i])
        break
      }
    }
  }

  return ans
};

console.log(stringMatching(["leetcode","et","code"]))

File: go-archive/string-matching-in-an-array.go
func stringMatching(words []string) []string {
    ans:=[]string{}

    for i:=0; i < len(words); i++ {
        for j:=0; j < len(words); j++ {
            if i == j {continue}
            if words[j] != strings.Join(strings.Split(words[j], words[i]), "") {
                ans = append(ans, words[i])
                break
            }
        }
    }

    return ans
}